### DISCUSSION TITLE
How to link with C static library ?

### DISCUSSION THREAD

**Discussion by leonsal** [vlang/v#18557] [Questions and Answers] [answered]
The documentation explains how to link with C dynamic libraries by using for example:

`#flag -lsqlite`
`#include "sqlite3.h"`
`#flag -L lib_path`

How to link with a static C library ?
Does it need to be passed through the command line ?
Regards.

**Comment by kevinmatthes** [comment, marked as answer]
@leonsal

I think I found a solution; at least I got the following small example working:

```c
// test.c
int test(void) {
    return 42;
}
```

```c
// test.h
extern int test(void);
```

Now, I compiled and bound `test.c` as follows:

```bash
gcc -c -std=c99 -Wall -Werror -Wextra -Wpedantic test.c
ar rs libtest.a test.o
```

To make sure that neither the C file nor its object code would be called from V, I removed both the `test.c` and `test.o`.

```v
// test.v
#flag -ltest
#flag -L.
#flag -I .
#include "test.h"

fn C.test() int

fn main() {
        println('${C.test()}')
}
```

And finally, I compiled the `test.v` with `v test.v` which gave me the executable `./test`; running `./test` will print the expectation:  42.

**Reply by kevinmatthes** [reply]
@leonsal

Compared to your suggestion in your initial comment, I needed to add the include flag `-I .` to ensure the header file being found.  Apart from that detail, the small example I posted looks similar to the documentation.  I assume that this include flag should be the major difference when calling a static library rather to a dynamic / shared one.

**Reply by JalonSolov** [reply]
Since you're not doing a `#include` in the .c file, the C compiler doesn't look for the .h file at all.  If you did have a `#include <somefile.h>`, you might need an extra compiler flag there, as well.
